snapd-glib/snapd-client.c

```diff
@@ -2443,16 +2462,22 @@ snapd_client_connect_sync (SnapdClient *client,
  */
 void
 snapd_client_connect_async (SnapdClient *client,
-                            GCancellable *cancellable, GAsyncReadyCallback callback, gpointer user_data)
+                            GCancellable *cancellable,
+                            GAsyncReadyCallback callback,
+                            gpointer user_data)
 {
-    GTask *task;
-    g_autoptr(GError) error = NULL;
+    SnapdClientPrivate *priv;
+    g_autoptr(GTask) task = NULL;
+
+    g_return_if_fail (SNAPD_IS_CLIENT (client));
+
+    priv = snapd_client_get_instance_private (client);
 
+    g_main_context_push_thread_default (priv->context);
```

Comment:
Here, the task's callback will be called in the thread-default main context, so we need to change it.